File: Computer-Science-50-Harward-University-edX/pset6/vigenere.py
import sys
import cs50


def main():

    # checking whether number of arguments is correct
    if len(sys.argv) != 2:
        print("Wrong number of arguments!")
        exit(1)

    # extracts integer from input
    key = sys.argv[1]

    if not key.isalpha():
        print("Wrong key!")
        exit(2)

    # text input from user
    text = cs50.get_string("plaintext:  ")

    print("ciphertext: ", end = "")

    # cursor for key
    cursor = 0

    # iterating over all characters in string
    for letter in text:

        # if character is alphabetical:
        if letter.isalpha():

            # gets number for encryption from key
            number = ord(key[cursor % len(key)].upper()) - ord('A')
            cursor += 1

            # if character is uppercase:
            if letter.isupper():
                print(chr((ord(letter) - ord('A') +  number) % 26 + ord('A')), end = "")

            # if character is lowercase:
            else:
                print(chr((ord(letter) - ord('a') +  number) % 26 + ord('a')), end = "")

        # if character is non-alphabetical:
        else:
            print(letter, end = "")
    # new line
    print()

    # great success
    exit(0)


# executes function
if __name__ == "__main__":
     main()

File: Computer-Science-50-Harward-University-edX/pset2/vigenere.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>
#include <cs50.h>


int main(int argc, string argv[])
{

    // checking whether number of arguments is correct
    if (argc != 2)
    {
        printf("Wrong number of arguments!");
        // if number incorrect, quits the program with error code '1'
        return 1;
    }

    // saving key from input
    string key = argv[1];

    // checking whether key is only alphabetical
    for (int i = 0, n = strlen(key); i < n; i++)
    {
        // checking whether all character in  key are alphabetical
        if (!isalpha(key[i]))
        {
            printf("Wrong key!");
            // if any non-alphabetical character, quits the program with error code '1'
            return 1;
        }
    }


    // text input from user
    string text = get_string("plaintext:  ");

    printf("ciphertext: ");

    // iterating over all characters in string
    for (int i = 0, n = strlen(text), j = 0; i < n; i++)
    {
        char character = text[i];
        int number;


        // if character is uppercase:
        if (isupper(character))
        {
            // getting cipher number from key word
            number = toupper(key[j % strlen(key)]) - 'A';
            j += 1;
            printf("%c", (character - 'A' + number) % 26 + 'A');
        }

        // if character is lowercase:
        else if (islower(character))
        {
            // getting cipher number from key word
            number = toupper(key[j % strlen(key)]) - 'A';
            j += 1;
            printf("%c", (character - 'a' + number) % 26 + 'a');
        }

        // if character is non-alphabetical:
        else
        {
            printf("%c", character);
        }

    }
    printf("\n");
    // quits the program with no error code
    return 0;
}
